### PR TITLE
Removed path to datadir in exception

### LINKED ISSUE #2046: Stop logging my data directory path in home page in case of error
Hello,
honestly I have no idea what to do with issues that I've opened for owncloud but that also affect nextcloud directly.
I will replicate https://github.com/owncloud/core/issues/26386 here, fell free to close it if you have already a plan to port issues from there automatically.

### Steps to reproduce
1. store the user data on an external hard disk
2. remove the hard disk or unmount it
3. open the oc home page
### Expected behaviour

It should tell me there is a configuration error
### Actual behaviour

The home page (available to everyone, logged or anonymous users) tells me there is a configuration error (the Data Directory is invalid) BUT it also tells everyone the path of the data directory. This is both a privacy and security concerns. I'm not sure if intentional or not, but I would disable the print of the full path in home page. I could not find any configuration how to disable this behaviour

```
Data directory (/full/path/to/data) is invalid
Please check that the data directory contains a file ".ocdata" in its root.
```
### Server configuration

**Operating system**: 
debian

**Web server:** 
apache

**Database:** 
mysql

**PHP version:** 
5

**ownCloud version:** (see ownCloud admin page)
9.1

**Updated from an older ownCloud or fresh install:**
fresh install
**Where did you install ownCloud from:**
From the repository
**Signing status (ownCloud 9.0 and above):**

**The content of config/config.php:**

```
$CONFIG = array (
   'instanceid' => 'my.id',
   'passwordsalt' => 'pepper',
   'secret' => 'my.secret',
   'trusted_domains' =>
   array (
       0 => 'my.oc.com',
   ),
   'datadirectory' => '/full/path/to/data',
   'overwrite.cli.url' => 'https://my.oc.com',
   'dbtype' => 'mysql',
   'version' => '9.1.0.15',
   'dbname' => 'owncloud',
   'dbhost' => 'localhost',
   'dbtableprefix' => 'own_',
   'dbuser' => 'mydbuser',
   'dbpassword' => '***REMOVED SENSITIVE VALUE***',
   'logtimezone' => 'Europe/Paris',
   'logfile' => '/var/log/owncloud.log',
   'loglevel' => '2', // Warning
   'installed' => true,
   'cron_log' => true,
   'memcache.local' => '\OC\Memcache\APCu',
 );
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
Local on an external hard disk
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### PR DESCRIPTION
**Pull request by MorrisJobke**
* downstream of https://github.com/owncloud/core/pull/26643
* fixed #2046

### CODE REVIEW

**Review by rullzer — APPROVED** [review]
Fine by me

File: lib/private/legacy/util.php [hunk 1]
```diff
@@ -698,22 +698,22 @@ public static function checkServer(\OC\SystemConfig $config) {
...
 			} else if (!is_writable($CONFIG_DATADIRECTORY) or !is_readable($CONFIG_DATADIRECTORY)) {
 				//common hint for all file permissions error messages
 				$permissionsHint = $l->t('Permissions can usually be fixed by '
 					. '%sgiving the webserver write access to the root directory%s.',
-					array('<a href="' . $urlGenerator->linkToDocs('admin-dir_permissions') . '" target="_blank" rel="noreferrer">', '</a>'));
-				$errors[] = array(
-					'error' => 'Data directory (' . $CONFIG_DATADIRECTORY . ') not writable',
+					['<a href="' . $urlGenerator->linkToDocs('admin-dir_permissions') . '" target="_blank" rel="noreferrer">', '</a>']);
+				$errors[] = [
+					'error' => 'Your Data directory is not writable',
```
Comment (nickvergessen) [inline lib/private/legacy/util.php]: lowercase `data` ?

File: lib/private/legacy/util.php [hunk 2]
```diff
@@ -928,10 +928,10 @@ public static function checkDataDirectoryPermissions($dataDirectory) {
 			clearstatcache();
 			$perms = substr(decoct(@fileperms($dataDirectory)), -3);
 			if (substr($perms, 2, 1) != '0') {
-				$errors[] = array(
-					'error' => $l->t('Data directory (%s) is readable by other users', array($dataDirectory)),
+				$errors[] = [
+					'error' => $l->t('Your Data directory is readable by other users'),
```
Comment (nickvergessen) [inline lib/private/legacy/util.php]: lowercase `data`

File: lib/private/legacy/util.php [hunk 3]
```diff
@@ -949,13 +949,13 @@ public static function checkDataDirectoryValidity($dataDirectory) {
 		$errors = [];
 		if ($dataDirectory[0] !== '/') {
 			$errors[] = [
-				'error' => $l->t('Data directory (%s) must be an absolute path', [$dataDirectory]),
+				'error' => $l->t('Your Data directory must be an absolute path'),
```
Comment (nickvergessen) [inline lib/private/legacy/util.php]: lowercase `data`

File: lib/private/legacy/util.php [hunk 4]
```diff
@@ -949,13 +949,13 @@ public static function checkDataDirectoryValidity($dataDirectory) {
...
 				'hint' => $l->t('Check the value of "datadirectory" in your configuration')
 			];
 		}
 		if (!file_exists($dataDirectory . '/.ocdata')) {
 			$errors[] = [
-				'error' => $l->t('Data directory (%s) is invalid', [$dataDirectory]),
+				'error' => $l->t('Your Data directory  is invalid'),
```
Comment (nickvergessen) [inline lib/private/legacy/util.php]: lowercase `data`

File: lib/private/legacy/util.php (hunk 1 above)
Comment (nickvergessen) [inline lib/private/legacy/util.php]: lowercase `data`

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**